tests/debugger/test_debugger_base.py

```diff
@@ -2,32 +2,63 @@
 # This product includes software developed at Datadog (https://www.datadoghq.com/).
 # Copyright 2021 Datadog, Inc.
 
-from utils import scenarios, interfaces, weblog, features, missing_feature, context
+from utils import interfaces
 from utils.tools import logger
-
-
-def validate_probes(expected_probes):
-    def get_probes_map():
-        agent_logs_endpoint_requests = list(interfaces.agent.get_data(path_filters="/api/v2/logs"))
-        probe_hash = {}
-
-        for request in agent_logs_endpoint_requests:
-            content = request["request"]["content"]
-            if content is not None:
-                for content in content:
+from packaging import version
+import json
+import re
+
+_CONFIG_PATH = "/v0.7/config"
+_DEBUGER_PATH = "/api/v2/debugger"
+_LOGS_PATH = "/api/v2/logs"
+_TRACES_PATH = "/api/v0.2/traces"
+
+
+def read_data():
+    tracer = list(interfaces.library.get_data(_CONFIG_PATH))[0]["request"]["content"]["client"]["client_tracer"]
+
+    if tracer["language"] == "java":
+        tracer_version = version.parse(re.sub(r"[^0-9.].*$", "", tracer["tracer_version"]))
+        if tracer_version > version.parse("1.27.0"):
+            path = _DEBUGER_PATH
+        else:
+            path = _LOGS_PATH
+    else:
+        path = _LOGS_PATH
+
+    return list(interfaces.agent.get_data(path))
+
+
+def get_probes_map(data_set):
+    probe_hash = {}
+
+    for data in data_set:
+        contents = data["request"].get("content", [])
+        if contents is not None:
+            for content in contents:
+                if "content" in content:
+                    d_contents = json.loads(content["content"])
+                    for d_content in d_contents:
+                        debugger = d_content["debugger"]
+                        if "diagnostics" in debugger:
+                            probe_id = debugger["diagnostics"]["probeId"]
+                            probe_hash[probe_id] = debugger["diagnostics"]
```

Comment:
Do you think there is a way to simplify this?

Reply:
I tried to make it more readable here: 0c1d33a227da89ed0e810f7a8945934313c54181